tv_apps/tv-deck/js/tv_deck.js

```diff
@@ -290,57 +293,27 @@
     this.panelTimeoutId = null;
   };
 
-    /**
-   * Update pin button in button-group-panel and contextmenu.
-   */
-  proto.updatePinButton = function td_updatePinButton() {
-    if (this.pinCard.pinnedChannels[this.channelManager.currentHash]) {
-      // Show unpin button if current channel is pinned.
-      this.pinButtonContextmenu.setAttribute('data-l10n-id', 'unpin-from-home');
-      this.pinButtonContextmenu.onclick = this._unpinFromHome.bind(this);
-      this.pinButton.onclick = this._unpinFromHome.bind(this);
+  proto._returnMeta = function cm__returnMeta(evt) {
+    var port = IACHandler.getPort('tvdeck-getmeta');
+    if (this.selfApp) {
+      port.postMessage({
+        origin: this.selfApp.origin,
+        manifestURL: this.selfApp.manifestURL
+      });
     } else {
-      // Show pin button if current channel is not pinned yet.
-      this.pinButtonContextmenu.setAttribute('data-l10n-id', 'pin-to-home');
-      this.pinButtonContextmenu.onclick = this._pinToHome.bind(this);
-      this.pinButton.onclick = this._pinToHome.bind(this);
+      navigator.mozApps.getSelf().onsuccess = function(evt){
```

Comment:
Do you have a STR to reproduce this case? The first start-up? If yes, please write some comments here.